flows/download_catalog.py

```diff
@@ -294,62 +330,85 @@ def query_all(coo_centre, radius=24*u.arcmin, dist_cutoff=2*u.arcsec):
 		list: List of dicts with catalog stars.
 
 	.. codeauthor:: Rasmus Handberg <rasmush@phys.au.dk>
+	.. codeauthor:: Emir Karamehmetoglu <emir.k@phys.au.dk>
 	"""
 
 	# Query the REFCAT2 catalog using CasJobs around the target position:
 	results = query_casjobs_refcat2(coo_centre, radius=radius)
+	AT_results = Table(results)
 
 	# Query APASS around the target position:
 	results_apass = query_apass(coo_centre, radius=radius)
+	AT_apass = Table(results_apass)
 
 	# Match the two catalogs using coordinates:
 	# https://docs.astropy.org/en/stable/coordinates/matchsep.html#matching-catalogs
-	ra = np.array([r['ra'] for r in results])
-	decl = np.array([r['decl'] for r in results])
-	refcat = SkyCoord(ra=ra, dec=decl, unit=u.deg, frame='icrs')
+	#ra = np.array([r['ra'] for r in results])
+	#decl = np.array([r['decl'] for r in results])
+	refcat = SkyCoord(ra=AT_results['ra'], dec=AT_results['decl'], unit=u.deg, frame='icrs')
 
-	ra_apass = np.array([r['ra'] for r in results_apass])
-	decl_apass = np.array([r['decl'] for r in results_apass])
-	apass = SkyCoord(ra=ra_apass, dec=decl_apass, unit=u.deg, frame='icrs')
+	#ra_apass = np.array([r['ra'] for r in results_apass])
+	#decl_apass = np.array([r['decl'] for r in results_apass])
+	apass = SkyCoord(ra=AT_apass['ra'], dec=AT_apass['decl'], unit=u.deg, frame='icrs')
 
 	# Match the two catalogs:
 	idx, d2d, _ = apass.match_to_catalog_sky(refcat)
+	sep_constraint = d2d <= dist_cutoff # Reject any match further away than the cutoff:
+	idx_apass = np.arange(len(idx)) # since idx maps apass to refcat
+
+	# Update results table with APASS bands of interest
+	AT_results['B_mag'][idx[sep_constraint]] = AT_apass[idx_apass[sep_constraint]]['B_mag']
+	AT_results['V_mag'][idx[sep_constraint]] = AT_apass[idx_apass[sep_constraint]]['V_mag']
+	AT_results['u_mag'][idx[sep_constraint]] = AT_apass[idx_apass[sep_constraint]]['u_mag']
+
+	# Create SDSS cat
+	AT_sdss = query_sdss(coo_centre, radius=radius, dr=16, clean=True)
+	sdss = coords.SkyCoord(ra=AT_sdss['ra'], dec=AT_sdss['dec'], unit=u.deg, frame='icrs')
+
+	# Match to dist_cutoff sky distance (angular) apart
+	idx, d2d, _ = sdss.match_to_catalog_sky(refcat)
+	sep_constraint = d2d <= dist_cutoff
+	idx_sdss = np.arange(len(idx)) # since idx maps sdss to refcat
+	# TODO: Maybe don't (potentially) overwrite APASS uband with SDSS uband. Decide which is better.
+	AT_results['u_mag'][idx[sep_constraint]] = AT_sdss[idx_sdss[sep_constraint]]['psfMag_u']
+
+	# # Go through the matches and make sure they are valid:
```

Comment:
Why is all of this uncommented? It is vital for the matching with APASS (and also SDSS) that this is still there to ensure proper input for the database

Reply:
It is redundant now. Using tables, we don't need to do it like this.